packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java

```diff
@@ -766,174 +776,13 @@ ResponseEntity<DocumentAsset> postPdfExtractions(
-			try {
-				final ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(zipFileResp.getBody());
-				final ZipInputStream zipInputStream = new ZipInputStream(byteArrayInputStream);
-
-				ZipEntry entry = zipInputStream.getNextEntry();
-				while (entry != null) {
-
-					fileMap.put(entry.getName(), zipEntryToHttpEntity(zipInputStream));
-
-					entry = zipInputStream.getNextEntry();
-				}
-
-				zipInputStream.closeEntry();
-				zipInputStream.close();
-			} catch (final IOException e) {
-				throw new RuntimeException("Unable to extract the contents of the zip file", e);
-			}
-
-			final ResponseEntity<JsonNode> textResp = extractionProxy.text(jobId);
-			if (!textResp.getStatusCode().is2xxSuccessful()) {
-				throw new RuntimeException("Unable to fetch the text extractions");
-			}
-
-			// clear existing assets
-			document.setAssets(new ArrayList<>());
-
-			for (final ExtractionAssetType extractionType : ExtractionAssetType.values()) {
-				final ResponseEntity<JsonNode> response = extractionProxy.extraction(jobId,
-						extractionType.toStringPlural());
-				log.info(" {} response status: {}", extractionType, response.getStatusCode());
-				if (!response.getStatusCode().is2xxSuccessful()) {
-					log.warn("Unable to fetch the {} extractions", extractionType);
-					continue;
-				}
-
-				for (final JsonNode record : response.getBody()) {
-
-					String fileName = "";
-					if (record.has("img_pth")) {
-
-						final String path = record.get("img_pth").asText();
-						fileName = path.substring(path.lastIndexOf("/") + 1);
-
-						if (fileMap.containsKey(fileName)) {
-							log.warn("Unable to find file {} in zipfile", fileName);
-						}
-
-						final HttpEntity file = fileMap.get(fileName);
-
-						documentService.uploadFile(documentId, fileName, file);
-
-					} else {
-						log.warn("No img_pth found in record: {}", record);
-					}
-
-					final DocumentExtraction extraction = new DocumentExtraction();
-					extraction.setFileName(fileName);
-					extraction.setAssetType(extractionType);
-					extraction.setMetadata(mapper.convertValue(record, Map.class));
-
-					document.getAssets().add(extraction);
-				}
-			}
-
-			String responseText = "";
-			for (final JsonNode record : textResp.getBody()) {
-				if (record.has("content")) {
-					responseText += record.get("content").asText() + "\n";
-				} else {
-					log.warn("No content found in record: {}", record);
-				}
-			}
-
-			document.setText(responseText);
-
-			// update the document
-			document = documentService.updateAsset(document).orElseThrow();
-
-			return ResponseEntity.ok(document);
-
-		} catch (final Exception e) {
-			final String error = "Unable to extract pdf";
-			log.error(error, e);
-			throw new ResponseStatusException(
-					org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR,
-					error);
-		}
+		final String currentUserId = currentUserService.get().getId();
+		extractionService.extractPDF(documentId, currentUserId);
```

Comment:
`extractionService.extractPDF` looks like it can throw `RuntimeException` if not other exceptions... can we put this in a try/catch and throw a `ResponseStatusException` for the front end to deal with?

Reply:
extractPDF should throw no Exceptions - I guess it the executor fails to take on the task...